src/cc_catalog_airflow/dags/provider_api_scripts/phylopic.py

```diff
@@ -0,0 +1,357 @@
+    IDs = _get_image_IDs(**args)
+
+    ctr = 1
+    while ctr <= len(IDs):
+        if IDs[ctr - 1] is not None:
+            details = _get_meta_data(IDs[ctr - 1])
+
+            if details is not None:
+                image_store.add_item(foreign_landing_url=details[1],
+                                     image_url=details[2],
+                                     thumbnail_url=details[3],
+                                     license_url=details[6],
+                                     width=details[4],
+                                     height=details[5],
+                                     creator=details[7],
+                                     title=details[8],
+                                     meta_data=details[9],
+                                     )
+
+        if ctr % LIMIT == 0:
+            # Add 5 images at a time to file
+            image_store.commit()
+            logger.info('Flushed current buffer to file.')
+        ctr += 1
+
+    image_store.commit()
+
+
+def _get_total_images():
+    # Get the total number of PhyloPic images
+    total = 0
+    endpoint = 'http://phylopic.org/api/a/image/count'
+    result = _get_response_json(endpoint=endpoint, retries=2)
+
+    if result and result.get('success') is True:
+        total = result.get('result')
+
+    return total
+
+
+def _get_image_IDs(**args):
+    limit = LIMIT
+    offset = 0
+    endpoint = ''
+
+    if args.get('date'):
+        # Get a list of recently updated/uploaded objects.
+        date = args['date']
+        endpoint = 'http://phylopic.org/api/a/image/list/modified/{}'.format(
+                  date)
+
+    else:
+        # Get all images and limit the results for each request.
+        offset = args['offset']
+        endpoint = 'http://phylopic.org/api/a/image/list/{}/{}'.format(
+                  offset, limit)
+
+    if endpoint == '':
+        logger.warning('Unable to parse endpoint: {}, args: {}'.format(
+                       endpoint, args))
+        return [None]
+
+    result = _get_response_json(endpoint=endpoint, retries=2)
+    image_IDs = []
+
+    if result and result.get('success') is True:
+        data = list(result.get('result'))
+
+        if len(data) > 0:
+            for i in range(len(data)):
+                image_IDs.append(data[i].get('uid'))
+
+    if not image_IDs:
+        logger.warning('No content available!')
+        return [None]
+
+    return image_IDs
+
+
+def _get_meta_data(_uuid):
+    logger.info('Processing UUID: {}'.format(_uuid))
+
+    start_time = time.time()
+    base_URL = 'http://phylopic.org'
+    img_URL = ''
+    thumbnail = ''
+    width = ''
+    height = ''
+    foreign_id = ''
+    foreign_url = ''
+    meta_data = {}
+    creator = ''
+    title = ''
+
+    endpoint = "http://phylopic.org/api/a/image/{}?options=credit+" \
+               "licenseURL+pngFiles+submitted+submitter+taxa+canonicalName" \
+               "+string+firstName+lastName".format(_uuid)
```

Comment:
(by passing it through `_get_response_json`, of course)

Reply:
I preferred to write it this way since this is more of a query string, rather than parameters. To use it as a parameters dict, there will only be one `options` key. 
So, that'll effectively make it just one parameter.
It'll look something like:
 ```params = {'options': "credit licenseURL pngFiles submitted submitter taxa canonicalName string firstName lastName"}```.
Do you want me to write it this way, or I stick to the current format?

Reply:
I think I'm fine with leaving it this way for now.